src/test/spec/unified_runner/operation.rs

```diff
@@ -0,0 +1,801 @@
+use std::{collections::HashMap, fmt::Debug, ops::Deref};
+
+use async_trait::async_trait;
+use futures::stream::TryStreamExt;
+use serde::Deserialize;
+
+use super::{Entity, ExpectError, Operation};
+
+use crate::{
+    bson::{doc, Bson, Deserializer as BsonDeserializer, Document},
+    error::Result,
+    options::{
+        AggregateOptions,
+        CountOptions,
+        DeleteOptions,
+        DistinctOptions,
+        EstimatedDocumentCountOptions,
+        FindOneAndDeleteOptions,
+        FindOneAndReplaceOptions,
+        FindOneAndUpdateOptions,
+        FindOneOptions,
+        FindOptions,
+        InsertManyOptions,
+        InsertOneOptions,
+        ListCollectionsOptions,
+        ListDatabasesOptions,
+        ReplaceOptions,
+        UpdateModifications,
+        UpdateOptions,
+    },
+    test::util::EventClient,
+    Collection,
+    Database,
+};
+
+#[async_trait]
+pub trait TestOperation: Debug {
+    /// The command names to monitor as part of this test.
+    fn command_names(&self) -> &[&str];
+
+    // The linked issue causes a warning that cannot be suppressed when providing a default
+    // implementation for these functions.
+    // <https://github.com/rust-lang/rust/issues/51443>
+    async fn execute_on_collection(&self, collection: &Collection) -> Result<Option<Bson>>;
+
+    async fn execute_on_client(&self, client: &EventClient) -> Result<Option<Bson>>;
+
+    async fn execute_on_database(&self, database: &Database) -> Result<Option<Bson>>;
+}
+
+#[derive(Debug)]
+pub struct EntityOperation {
```

Comment:
This file is a copy of `operation.rs` from the v2 runner -- the only difference is the definition of `EntityOperation` here to accommodate for the new fields defined in the unified runner. I didn't bother trying to make them share code since I anticipate we'll be removing that runner as we port tests over to the unified runner anyway.